doc/elasticdl_on_sqlflow.md

```diff
@@ -0,0 +1,141 @@
+# _Design:_ ElasticDL on sqlflow
+
+## Overview
+
+This is a design doc on integration with [ElasticDL](https://github.com/wangkuiyi/elasticdl).
+
+### User Interface
+
+#### Training Job Submission
+
+```sql
+SELECT 
+    c1, c2, c3, c4, c5 as class
+FROM training_data
+TRAIN ElasticDLKerasClassifier
+WITH
+  optimizer = "optimizer"
+  loss = "loss"
+  eval_metrics = "eval_metrics_fn"
+  num_classes = 10
+COLUMN
+  c1,
+  NUMERIC(c2, 10),
+  BUCKET(c3, [0, 10, 100]),
+  c4
+LABEL class
+INTO trained_elasticdl_keras_classifier;
+```
+
+#### Prediction Job Submission
+
+```sql
+SELECT 
+    c1, c2, c3, c4
+FROM prediction_data
+PREDICT prediction_results_table
+WITH
+  num_classes = 10
+USING trained_elasticdl_keras_classifier;
+```
+
+#### Run-time Configurations
+
+Users can provide run-time configurations to ElasticDL job via additional parameters with prefix "runtime" within `WITH` clause, for example:
+
+```sql
+SELECT
+    c1, c2, c3, c4, c5 as class
+FROM training_data
+TRAIN ElasticDLKerasClassifier
+WITH
+  optimizer = "optimizer"
+  loss = "loss"
+  eval_metrics = "eval_metrics_fn"
+  num_classes = 10
+  runtime.num_epochs = 2
```

Comment:
FYI, in the case of ALPS as backend, a user would write`train.num_epochs`.

Reply:
Isn’t that “train” redundant with the “TRAIN” clause?

Reply:
The goal is to differentiate with other model-specific parameters, similar to Estimator’s RunConfig.